python/hidet/graph/transforms/conv_channel_last.py

```diff
@@ -0,0 +1,287 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Dict, Tuple, Iterable, Optional
+
+from hidet.graph.operator import Operator, Tensor
+from hidet.graph.flow_graph import FlowGraph
+from hidet.graph.transforms import GraphPass
+from hidet.graph.graph_utils.functors import analyze_usage
+
+
+class PermutedOp:
+    """
+    The `reforward()` method of a PermutedOp should accept a tensor map, create channel last inputs
+    as needed, produce channel last outputs, and update the tensor map with new inputs and outputs.
+    The base case of PermutedOp can accept channel last inputs by updating its 'axis' attribute.
+    """
+
+    # Some imports will cause circular import errors if imported at top level of class
+    # Therefore we import those inside PermutedOp.initialize_scoped_ops() instead.
+    from hidet.graph.ops.arithmetic import UnaryElementwiseOp, BinaryElementwiseOp
+    from hidet.graph.ops.transform import ConcatOp
+
+    regular_operators = (UnaryElementwiseOp, BinaryElementwiseOp, ConcatOp)
+    channel_last_operators = tuple()
+    scoped_operators = tuple()
+
+    def __init__(self, op: Operator) -> None:
+        self.op = op
+
+    def reforward(self, tensor_map) -> None:
+        from hidet.graph.ops.transform import transpose
+
+        node = self.op
+        new_inputs: List[Tensor] = []
+        update_attributes = {}
+        new_perms = []
```

Comment:
Better to add the type hints for `tensor_map`, `update_attributes` and `new_perms` to reduce the time to learn the code when people reading the code.